### PR TITLE
Revise dwarf chapter

### PR DESCRIPTION
**Pull request by kito-cheng**
Changes:
- Add new chapter to describe RISC-V using std DWARF spec. 
- Add introduction sentence to DWARF register number mapping table

### CODE REVIEW

**Comment by kito-cheng** [comment]
@cmuellner

File: riscv-dwarf.adoc [hunk 1]
```diff
@@ -4,8 +4,17 @@ ifeval::["{docname}" == "riscv-dwarf"]
 include::prelude.adoc[]
 endif::[]
 
+== DWARF Debugging Format
+
+DWARF debugging format for RISC-V follows the
+https://dwarfstd.org/[standard DWARF specification], this specification only
+describe RISC-V specific definitions.
```
Comment (jrtc27) [inline riscv-dwarf.adoc:11]:
```suggestion
The DWARF debugging format for RISC-V follows the
https://dwarfstd.org/[standard DWARF specification]; this specification only
describes RISC-V-specific definitions.
```
though do we need to use a better way of referencing it than an inline link?

**Comment by kito-cheng** [comment]
@jrtc27 

> though do we need to use a better way of referencing it than an inline link?

Maybe using footnote to doing that? I could create a separated PR for that, and also convert all other inline link.

**Comment by cmuellner** [comment]
> > though do we need to use a better way of referencing it than an inline link?
> 
> Maybe using footnote to doing that? I could create a separated PR for that, and also convert all other inline link.

As a rule of thumb, we can follow the unpriv spec or do something better.
And that document has a `Bibliography` at the end.
What could be done better: use [MEANINGFUL NAME] instead of [NUMBER] as an identifier.

The rest looks good.

**Comment by jrtc27** [comment]
> > > though do we need to use a better way of referencing it than an inline link?
> > 
> > 
> > Maybe using footnote to doing that? I could create a separated PR for that, and also convert all other inline link.
> 
> As a rule of thumb, we can follow the unpriv spec or do something better. And that document has a `Bibliography` at the end. What could be done better: use [MEANINGFUL NAME] instead of [NUMBER] as an identifier.

I guess we should be using https://docs.asciidoctor.org/asciidoc/latest/sections/bibliography/ rather than hand-rolling things. The numbers are the default from bibtex, unlikely a conscious choice on the unprivileged spec authors' parts (and documents that _don't_ use numbers always look odd to me as it's unusual in our field...).

**Comment by cmuellner** [comment]
> > > > though do we need to use a better way of referencing it than an inline link?
> > > 
> > > 
> > > Maybe using footnote to doing that? I could create a separated PR for that, and also convert all other inline link.
> > 
> > 
> > As a rule of thumb, we can follow the unpriv spec or do something better. And that document has a `Bibliography` at the end. What could be done better: use [MEANINGFUL NAME] instead of [NUMBER] as an identifier.
> 
> I guess we should be using https://docs.asciidoctor.org/asciidoc/latest/sections/bibliography/ rather than hand-rolling things. The numbers are the default from bibtex, unlikely a conscious choice on the unprivileged spec authors' parts (and documents that _don't_ use numbers always look odd to me as it's unusual in our field...).

Certainly a matter of taste (and thus not worth to argue about).
But I would see both widely used in IT (non is unusual).
E.g. the link you provided uses named indices.

**Comment by jrtc27** [comment]
> > > > > though do we need to use a better way of referencing it than an inline link?
> > > > 
> > > > 
> > > > Maybe using footnote to doing that? I could create a separated PR for that, and also convert all other inline link.
> > > 
> > > 
> > > As a rule of thumb, we can follow the unpriv spec or do something better. And that document has a `Bibliography` at the end. What could be done better: use [MEANINGFUL NAME] instead of [NUMBER] as an identifier.
> > 
> > 
> > I guess we should be using https://docs.asciidoctor.org/asciidoc/latest/sections/bibliography/ rather than hand-rolling things. The numbers are the default from bibtex, unlikely a conscious choice on the unprivileged spec authors' parts (and documents that _don't_ use numbers always look odd to me as it's unusual in our field...).
> 
> Certainly a matter of taste (and thus not worth to argue about). But I would see both widely used in IT (non is unusual). E.g. the link you provided uses named indices.

Yeah, I don't particularly mind, if the AsciiDoc default is to use the names then we should just use names. That seems to be standard practice outside the LaTeX world.

**Comment by kito-cheng** [comment]
https://docs.asciidoctor.org/asciidoc/latest/sections/bibliography/ I tried this and seems work fine, but I would prefer to having a separated PR for convert those bibliography entries, other than that, is this PR ok for you?

**Review by jrtc27 — APPROVED** [review]
(but please squash before merging, and you'll also have to rebase the bibliography PR afterwards)

**Pull request merged**